lib/puppet/functions/loadjson.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# @summary
+#   Load a JSON file containing an array, string, or hash, and return the data
+#   in the corresponding native data type.
+#
+# @example Example Usage:
+#   $myhash = loadjson('/etc/puppet/data/myhash.json')
+#   $myhash = loadjson('https://example.local/my_hash.json')
+#   $myhash = loadjson('https://username:password@example.local/my_hash.json')
+#   $myhash = loadjson('no-file.json', {'default' => 'value'})
+Puppet::Functions.create_function(:loadjson) do
```

Comment:
At the same time, shouldn't we namespace this function and create a (deprecated) shim like we started with a lot of the other functions?

Reply:
I didn't consider this.